HW0/P6/P6.txt

```diff
@@ -1 +1,121 @@
 # Your answers here
+
+Part P6A
+
+Example Output
+Output 1:
+
+Hi Job 0
+Hi Job 1
+Hi Job 2
+Hi Job 3
+Bye Job 3
+Bye Job 1
+Bye Job 2
+Hi Job 4
+Hi Job 5
+Hi Job 6
+Bye Job 0
+Hi Job 7
+Bye Job 4
+Hi Job 8
+Bye Job 6
+Hi Job 9
+Bye Job 5
+Bye Job 7
+Bye Job 8
+Bye Job 9
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+Output 2:
+
+Hi Job 0
+Hi Job 1
+Hi Job 2
+Hi Job 3
+Bye Job 2
+Bye Job 1
+Bye Job 3
+Hi Job 4
+Bye Job 0
+Hi Job 5
+Hi Job 6
+Hi Job 7
+Bye Job 4
+Bye Job 5
+Hi Job 8
+Hi Job 9
+Bye Job 7
+Bye Job 6
+Bye Job 8
+Bye Job 9
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+Output 3:
+
+Hi Job 0
+Hi Job 1
+Hi Job 2
+Hi Job 3
+Bye Job 0
+Hi Job 4
+Bye Job 1
+Hi Job 5
+Bye Job 3
+Bye Job 2
+Hi Job 6
+Hi Job 7
+Bye Job 4
+Bye Job 5
+Bye Job 6
+Bye Job 7
+Hi Job 8
+Hi Job 9
+Bye Job 9
+Bye Job 8
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+The results are odd becuase the jobs don't all start and stop in a
+consistent or symmetrical manner.  These runs all start with jobs 0-3 but
+they don't finish in the same order every time. This shows a latency
+issue. Some jobs are postponed because other operations are running.
+
+This could affect how we program in parallel because there are three
+paths (more computers, multicore, GPU) with different strengths so we
+have to weigh the costs and benefits of each.  We might also want to
+set it up differently and change the process size depending on what we are
+trying to compute and if it needs to be synchronized processing. We would
+need it if one part running depended on the other parts running as well,
+in other words, if they were interdependent.
```

Comment:
You explain the theoretical concept quite well but a practical example would help your case as well.